test/datasets/test_amazonreviewfull.py

```diff
@@ -0,0 +1,79 @@
+import os
+import tarfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.amazonreviewfull import AmazonReviewFull
+
+from ..common.case_utils import TempDirMixin, zip_equal, get_random_unicode
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "AmazonReviewFull")
+    temp_dataset_dir = os.path.join(base_dir, "temp_dataset_dir")
+    os.makedirs(temp_dataset_dir, exist_ok=True)
+
+    seed = 1
+    mocked_data = defaultdict(list)
+    for file_name in ("train.csv", "test.csv"):
+        txt_file = os.path.join(temp_dataset_dir, file_name)
+        with open(txt_file, "w", encoding="utf-8") as f:
+            for i in range(5):
+                label = seed % 5 + 1
+                rand_string = get_random_unicode(seed)
+                dataset_line = (label, f"{rand_string} {rand_string}")
+                # append line to correct dataset split
+                mocked_data[os.path.splitext(file_name)[0]].append(dataset_line)
+                f.write(f'"{label}","{rand_string}","{rand_string}"\n')
+                seed += 1
+
+    compressed_dataset_path = os.path.join(
+        base_dir, "amazon_review_full_csv.tar.gz"
+    )
+    # create tar file from dataset folder
+    with tarfile.open(compressed_dataset_path, "w:gz") as tar:
+        tar.add(temp_dataset_dir, arcname="amazon_review_full_csv")
+
+    return mocked_data
+
+
+class TestAmazonReviewFull(TempDirMixin, TorchtextTestCase):
```

Comment:
AmazonReviewFull and AmazonReviewPolarity are already merged into single Test case, so we no longer need this file :)

Reply:
Oops -- looks like my merge conflict skills are a bit rusty. 😅